pandaserver/dataservice/setupper_atlas_plugin.py

```diff
@@ -0,0 +1,2226 @@
+                                    file.md5sum = file.md5sum.strip()
+                                if file.checksum is not None:
+                                    file.checksum = file.checksum.strip()
+                        else:
+                            if job.prodSourceLabel not in ["managed", "test"]:
+                                add_to_lfn_map = False
+                        # check missing file
+                        if file.GUID == "NULL" or job.prodSourceLabel in [
+                            "managed",
+                            "test",
+                        ]:
+                            if file.lfn not in val_map:
+                                # append job to waiting list
+                                err_msg = f"GUID for {file.lfn} not found in rucio"
+                                self.logger.error(err_msg)
+                                file.status = "missing"
+                                if job not in jobs_failed:
+                                    job.jobStatus = "failed"
+                                    job.ddmErrorCode = ErrorCode.EC_GUID
+                                    job.ddmErrorDiag = err_msg
+                                    jobs_failed.append(job)
+                                    is_failed = True
+                                continue
+                        # add to all_lfns/all_guids
+                        if add_to_lfn_map:
+                            all_lfns.setdefault(job.getCloud(), [])
+                            all_guids.setdefault(job.getCloud(), [])
+                            all_scopes.setdefault(job.getCloud(), [])
+                            all_lfns[job.getCloud()].append(file.lfn)
+                            all_guids[job.getCloud()].append(file.GUID)
+                            all_scopes[job.getCloud()].append(file.scope)
+                # modify jobParameters
+                if not is_failed:
+                    for patt, repl in replace_list:
+                        job.jobParameters = re.sub(f"{patt} ", f"{repl} ", job.jobParameters)
+                    # append job to processed list
+                    jobs_processed.append(job)
+        # return if TA only
+        if self.only_ta:
+            self.logger.debug("end TA sessions")
+            return
+        # set data summary fields
+        for tmp_job in self.jobs:
+            try:
+                # set only for production/analysis/test
+                if tmp_job.prodSourceLabel not in ["managed", "test", "user", "prod_test"] + JobUtils.list_ptest_prod_sources:
+                    continue
+                # loop over all files
+                tmp_job.nInputDataFiles = 0
+                tmp_job.inputFileBytes = 0
+                tmp_input_file_project = None
+                tmp_input_file_type = None
+                for tmp_file in tmp_job.Files:
+                    # use input files and ignore DBR/lib.tgz
```

Comment:
lib.tgz? Are these the user tarballs? A comment would be appreciated. 
DBR/ddo see above.

Reply:
lib.tgz = user sandbox